tools/resourcedocsgen/cmd/metadata.go

```diff
@@ -208,12 +208,28 @@ func PackageMetadataCmd() *cobra.Command {
 			for _, requiredFile := range requiredFiles {
 				requiredFilePath := fmt.Sprintf("https://raw.githubusercontent.com/%s/%s/docs/%s",
 					repoSlug, version, requiredFile)
-				details, err := readRemoteFile(requiredFilePath, repoSlug.owner)
+				content, err := readRemoteFile(requiredFilePath, repoSlug.owner)
 				if err != nil {
 					return err
 				}
+				if len(content) == 0 {
+					continue
+				}
+
+				// Normalize end of line representation
+				content = bytes.ReplaceAll(content, []byte("\r\n"), []byte("\n"))
+
+				if rest, ok := bytes.CutPrefix(bytes.TrimLeft(content, "\n\t\r "), []byte("---\n")); ok {
+					content = append([]byte(`---
+# WARNING: this file was fetched from `+requiredFilePath+`
+# Do not edit by hand unless you're certain you know what you are doing!
+`), rest...)
+				} else {
+					return fmt.Errorf(`expected file %s to start with YAML front-matter ("---\n"), found leading %q`,
+						requiredFilePath, string(content))
```

Comment:
That could be a _lot_ of content; would it make sense to truncate?

Reply:
Yes, yes it would.

I had it truncated earlier, but changed it when I was debugging another issue. Thanks for catching that!